File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/npc/NpcDialogReplyMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait NpcDialogReplyMessage extends MessageDeserializer {
  val messageId = 5616

  protected trait Message extends BaseMessage with Serializable {
    val messageId = NpcDialogReplyMessage.messageId

    val replyId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(replyId)
    }

  }

  protected def create(replyId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val replyId = buf.readShort()

    create(replyId)
  }

}

object NpcDialogReplyMessage extends NpcDialogReplyMessage {
  case class MessageImpl(replyId: Short) extends Message
  type Target = MessageImpl

  override def create(replyId: Short) = MessageImpl(replyId)

  def apply(replyId: Short) = create(replyId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/NpcDialogReplyMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class NpcDialogReplyMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5616;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short replyId;
    
    public NpcDialogReplyMessage() { }
    
    public NpcDialogReplyMessage(short replyId) {
        this.replyId = replyId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(replyId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        replyId = buf.readShort();
        if (replyId < 0)
            throw new RuntimeException("Forbidden value on replyId = " + replyId + ", it doesn't respect the following condition : replyId < 0");
    }
    
}
